astroplan/moon.py

```diff
@@ -0,0 +1,164 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+This version of the `moon` module contains temporary solutions to calculating
+the position of the moon using PyEphem, awaiting solutions to the astropy issue
+[#3920](https://github.com/astropy/astropy/issues/3920).
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import numpy as np
+from astropy.time import Time
+from astropy.coordinates import (SkyCoord, Longitude, Latitude, GCRS,
+                                 CartesianRepresentation)
+import astropy.units as u
+from astropy.utils.data import download_file
+
+__all__ = ["get_moon", "calc_moon_phase_angle", "calc_moon_illumination"]
+
+def get_spk_file():
+    """
+    Get the Satellite Planet Kernel (SPK) file `de430.bsp` from NASA JPL.
+
+    Download the file from the JPL webpage once and subsequently access a
+    cached copy. This file is ~120 MB, and covers years ~1550-2650 CE [1]_.
+
+    .. [1] http://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/aareadme_de430-de431.txt
+    """
+    de430_url = ('http://naif.jpl.nasa.gov/pub/naif/'
+                 'generic_kernels/spk/planets/de430.bsp')
+    return download_file(de430_url, cache=True, show_progress=True)
+
+def get_moon(time, location, pressure, use_pyephem=True):
+    """
+    Position of the Earth's moon.
+
+    Currently uses PyEphem to calculate the position of the moon by default
+    (requires PyEphem to be installed). Set ``use_pyephem`` to `False` to
+    calculate the moon position with jplephem (requires jplephem to be
+    installed).
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time` or see below
+        Time of observation. This will be passed in as the first argument to
+        the `~astropy.time.Time` initializer, so it can be anything that
+        `~astropy.time.Time` will accept (including a `~astropy.time.Time`
+        object).
+
+    location : `~astropy.coordinates.EarthLocation`
+        Location of the observer on Earth
+
```

Comment:
Need to add `pressure` here in the docstring.  Was following the docs in my ipython session and got an error.